### PR TITLE
Feature/integrate onboarding answers

### PR DESCRIPTION
**Pull request by CeesWang**
### What does this pull request aim to achieve? 💯

_Please be concise and link any related issue(s):_

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](https://github.com/FightPandemics/FightPandemics/blob/master/README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`** (need to have cloned the repo not forked).
- [x] This branch is rebased/merged with the **latest master**.
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".

### CODE REVIEW

File: client/src/pages/Feed.js [hunk 1]
```diff
@@ -416,6 +416,54 @@ const Feed = (props) => {
...
+              handleOnboardingOptions(option, "providers");
+            }
+          } else {
+            let option = filters[1].options[0];
+            handleOnboardingOptions(option, "providers");
+          }
+        }
+      }
+    }
+  }, [filters, handleChangeType, props.history.location.state]);
```
Comment (CeesWang) [inline client/src/pages/Feed.js:465]: I have no Idea why it keeps filling in these fields in this array when I commit I hope someone can let me know why this is the case.

Comment (idjevm) [reply client/src/pages/Feed.js:465]: Will take a look and fix if neeeded

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      "version": "3.2.2",
+      "resolved": "https://registry.npmjs.org/@ant-design/colors/-/colors-3.2.2.tgz",
+      "integrity": "sha512-YKgNbG2dlzqMhA9NtI3/pbY16m3Yl/EeWBRa+lB1X1YaYxHrxNexiQYCLTWO/uDvAjLFMEDU+zR901waBtMtjQ==",
+      "requires": {
+        "tinycolor2": "^1.4.1"
+      }
+    },
+    "@ant-design/icons": {
+      "version": "4.2.1",
+      "resolved": "https://registry.npmjs.org/@ant-design/icons/-/icons-4.2.1.tgz",
```
Comment (idjevm) [inline package-lock.json:15]: Remove this file entirely. No need to update it or add it on your commits...

File: client/src/pages/Feed.js [hunk 3]
```diff
@@ -277,13 +277,13 @@ const Feed = (props) => {
     }
   };
 
-  const handleChangeType = (e) => {
+  const handleChangeType = useCallback((e) => {
```
Comment (idjevm) [inline client/src/pages/Feed.js:280]: Is this `useCallback` also added automatically?

File: client/src/pages/Feed.js (hunk 1 above)
Comment (idjevm) [reply client/src/pages/Feed.js:465]: How should it be like?

File: client/src/pages/Feed.js (hunk 3 above)
Comment (CeesWang) [reply client/src/pages/Feed.js:280]: yes also automatically added

File: client/src/pages/Feed.js (hunk 1 above)
Comment (CeesWang) [reply client/src/pages/Feed.js:465]: it should be like     }, []);

Comment (idjevm) [reply client/src/pages/Feed.js]: Cool, I have resolved this cuz I want to merge this one asap

File: package-lock.json (hunk 2 above)
Comment (idjevm) [reply package-lock.json]: Also resolved this so I can merge this one asap

**Comment by ppagarawal** [comment]
> So overall it looks great, but I think the filters are not working still -> https://share.getcloudapp.com/qGuKrGqW
> 
> As you can see in the video, the filters appear on the filter section but they do not affect anything from the posts on the feed/ are not being applied? I am not sure. Would you mind to investigate @CeesWang ?

cc: @joshmorel

**Comment by joshmorel** [comment]
Thanks @CeesWang 

I'm wondering if we're applying the filters correctly with respect to question 1  @ppagarawal @manualzuru ? I thought these were just for our contact list not to actually filter the feed because there isn't a really good correspondence between answers to these questions and filter options:

image
image

I'll take a look at the code later today.

**Comment by CeesWang** [comment]
@idjevm Regarding the video. 
Screenshot from 2020-06-29 14-11-28
These are the posts that show up when going through the onboarding questions. As in the video.

Screenshot from 2020-06-29 14-11-38
These posts show up after removing medical supplies.

Screenshot from 2020-06-29 14-11-47
These posts show up after adding medical supplies back in. 

The weird thing is that first and last instance where the medical supplies and location are exactly the same, the api call is also exactly the same. 

How can two identical api calls return different posts ? Which makes me believe its a backend issue. @idjevm @ppagarawal @joshmorel

**Comment by CeesWang** [comment]
Regarding the video, it has to do with India not having medical supplies posts. There are only two posts in India and they are both information type.

**Pull request merged**